lib/c.c

```diff
@@ -166,19 +166,29 @@ void __str_base10(char *pb, int val)
 
 void __str_base8(char *pb, int val)
 {
-    int c = INT_BUF_LEN - 1;
-    while (c > 0) {
-        int v = val & 0x7;
+    int c = INT_BUF_LEN - 1, v;
+    /*
+     * Because every 3 binary digits can be converted
+     * to 1 octal digit, here performs the conversion
+     * 10 times, derived from 32 divided by 3.
+     *
+     * Finally, the remaining 2 bits are processed after
+     * the loop.
+     * */
+    for (int i = 0; i < (sizeof(int) << 3) / 3; i++) {
```

Comment:
Perhaps you can lift the rhs expression of condition expression out of loop to squeeze last bit of performance?